### PR TITLE
river: 0.1.0 -> 0.1.2

### PR DESCRIPTION
**Pull request by ankhers**
###### Motivation for this change

Upgrade river to the most recent version

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by ankhers** [comment]
I removed the overrides for specific `zig` and `wlroots` versions because it works with latest. Should I be pinning that to the latest versions to ensure compatibility or is this okay for now and we can fix if something breaks?

**Comment by ankhers** [comment]
It looks like it. I hadn't searched first. Sorry about that! Feel free to close this one if it would be better to use that one.

File: pkgs/applications/window-managers/river/default.nix [hunk 1]
```diff
@@ -1,31 +1,16 @@
...
-, pixman
-, udev
-, libevdev
-, libinput
-, libX11
-, libGL
-}:
+{ lib, stdenv, fetchFromGitHub, zig, wayland, pkg-config, scdoc, xwayland
+, wayland-protocols, wlroots, libxkbcommon, pixman, udev, libevdev, libinput
+, libX11, libGL }:
```
Comment (SuperSandro2000) [inline pkgs/applications/window-managers/river/default.nix:3]: Please revert that. One line per input is the preferred format.

File: pkgs/applications/window-managers/river/default.nix [hunk 2]
```diff
@@ -55,10 +40,9 @@ stdenv.mkDerivation rec {
     runHook postInstall
   '';
 
-  /*
-    Builder patch install dir into river to get default config
-    When installFlags is removed, river becomes half broken.
-    See https://github.com/ifreund/river/blob/7ffa2f4b9e7abf7d152134f555373c2b63ccfc1d/river/main.zig#L56
+  /* Builder patch install dir into river to get default config
+     When installFlags is removed, river becomes half broken.
+     See https://github.com/ifreund/river/blob/7ffa2f4b9e7abf7d152134f555373c2b63ccfc1d/river/main.zig#L56
```
Comment (SuperSandro2000) [inline pkgs/applications/window-managers/river/default.nix:45]:
```suggestion
     See https://github.com/riverwm/river/blob/7ffa2f4b9e7abf7d152134f555373c2b63ccfc1d/river/main.zig#L56
```

Comment (ankhers) [reply pkgs/applications/window-managers/river/default.nix]: Good catch on this!

**Pull request merged**